### ISSUE TITLE
Question: How can recover a deleted recording from file backups?

### ISSUE THREAD

**Issue by sparse91** [blindsidenetworks/scalelite#561]
Hello
Is there any way for recovering?

**Comment by mk-hs** [comment]
zip it up and put it into the spool directory

**Comment by sparse91** [comment]
I `tar cf`ed meeting directory and put it into spool but It didn't work. I get this error:
`
E, [2021-05-19T09:03:04.889299 #7] ERROR -- : Failed to import recording: undefined method `update!' for nil:NilClass
WARN -- : /srv/scalelite/lib/recording_importer.rb:33:in `block (2 levels) in import': undefined method `update!' for nil:NilClass (NoMetod Error)
`

**Comment by mk-hs** [comment]
check if you've tar'ed it correctly:
Folder: presentation/id-timestamp
tar cfv id-timestamp.tar presentation

**Comment by sparse91** [comment]
```
cd /backup/published/presentation/8c52bddbe096cbc3024a917ebd49833b69669943-1621415697668
tar cfv 8c52bddbe096cbc3024a917ebd49833b69669943-1621415697668.tar *
mv 8c52bddbe096cbc3024a917ebd49833b69669943-1621415697668.tar /storage/spool/

logs:

May 22 14:36:00 bbb docker[3971873]: presentation/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1621415697751/textfiles/slide-6.txt
May 22 14:36:00 bbb docker[3971873]: presentation/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1621415697751/textfiles/slide-1.txt
May 22 14:36:00 bbb docker[3971873]: presentation/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1621415697751/textfiles/slide-8.txt
May 22 14:36:00 bbb docker[3971873]: presentation/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1621415697751/textfiles/slide-2.txt
May 22 14:36:00 bbb docker[3971873]: presentation/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1621415697751/textfiles/slide-5.txt
May 22 14:36:00 bbb docker[3971873]: presentation/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1621415697751/textfiles/slide-4.txt
May 22 14:36:00 bbb docker[3971873]: presentation/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1621415697751/textfiles/slide-3.txt
May 22 14:36:00 bbb docker[3971873]: presentation/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1621415697751/slide-3.png
May 22 14:36:00 bbb docker[3971873]: presentation/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1621415697751/slide-4.png
May 22 14:36:00 bbb docker[3971873]: presentation_text.json
May 22 14:36:00 bbb docker[3971873]: shapes.svg
May 22 14:36:00 bbb docker[3971873]: slides_new.xml
May 22 14:36:00 bbb docker[3971873]: video/
May 22 14:36:00 bbb docker[3971873]: video/webcams.webm
May 22 14:36:00 bbb docker[3971873]: E, [2021-05-22T10:06:00.585154 #7] ERROR -- : Failed to import recording: undefined method `update!' for nil:NilClass
May 22 14:36:00 bbb docker[3971873]: W, [2021-05-22T10:06:00.637311 #7]  WARN -- : /srv/scalelite/lib/recording_importer.rb:33:in `block (2 levels) in import': undefined method `update!' for nil:NilClass (NoMethodError)
May 22 14:36:00 bbb docker[3971873]:         from /usr/lib/ruby/2.6.0/fileutils.rb:128:in `chdir'
May 22 14:36:00 bbb docker[3971873]:         from /usr/lib/ruby/2.6.0/fileutils.rb:128:in `cd'
May 22 14:36:00 bbb docker[3971873]:         from /srv/scalelite/lib/recording_importer.rb:19:in `block in import'
May 22 14:36:00 bbb docker[3971873]:         from /usr/lib/ruby/2.6.0/tmpdir.rb:93:in `mktmpdir'
May 22 14:36:00 bbb docker[3971873]:         from /srv/scalelite/lib/recording_importer.rb:18:in `import'
May 22 14:36:00 bbb docker[3971873]:         from /srv/scalelite/lib/tasks/recordings.rake:17:in `block (4 levels) in <main>'
May 22 14:36:00 bbb docker[3971873]:         from /srv/scalelite/lib/tasks/recordings.rake:16:in `each'
May 22 14:36:00 bbb docker[3971873]:         from /srv/scalelite/lib/tasks/recordings.rake:16:in `block (3 levels) in <main>'
May 22 14:36:00 bbb docker[3971873]:         from /srv/scalelite/vendor/bundle/ruby/2.6.0/gems/listen-3.1.5/lib/listen/event/config.rb:23:in `call'
May 22 14:36:00 bbb docker[3971873]:         from /srv/scalelite/vendor/bundle/ruby/2.6.0/gems/listen-3.1.5/lib/listen/event/processor.rb:115:in `_process_changes'
May 22 14:36:00 bbb docker[3971873]:         from /srv/scalelite/vendor/bundle/ruby/2.6.0/gems/listen-3.1.5/lib/listen/event/processor.rb:19:in `block in loop_for'
May 22 14:36:00 bbb docker[3971873]:         from /srv/scalelite/vendor/bundle/ruby/2.6.0/gems/listen-3.1.5/lib/listen/event/processor.rb:15:in `loop'
May 22 14:36:00 bbb docker[3971873]:         from /srv/scalelite/vendor/bundle/ruby/2.6.0/gems/listen-3.1.5/lib/listen/event/processor.rb:15:in `loop_for'
May 22 14:36:00 bbb docker[3971873]:         from /srv/scalelite/vendor/bundle/ruby/2.6.0/gems/listen-3.1.5/lib/listen/event/loop.rb:84:in `_wait_for_changes'
May 22 14:36:00 bbb docker[3971873]:         from /srv/scalelite/vendor/bundle/ruby/2.6.0/gems/listen-3.1.5/lib/listen/event/loop.rb:42:in `block in setup'
May 22 14:36:00 bbb docker[3971873]:         from /srv/scalelite/vendor/bundle/ruby/2.6.0/gems/listen-3.1.5/lib/listen/internals/thread_pool.rb:6:in `block in add'
```

**Comment by mk-hs** [comment]
The structure inside the tar needs to be presentation/id-timestamp
so copy /backup/published/presentation/8c52bddbe096cbc3024a917ebd49833b69669943-1621415697668
to ~/presentation/8c52bddbe096cbc3024a917ebd49833b69669943-1621415697668

tar cfv 8c52bddbe096cbc3024a917ebd49833b69669943-1621415697668.tar ~/presentation

**Comment by sparse91** [comment]
It works. Thank you
